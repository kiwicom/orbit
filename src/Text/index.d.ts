// @flow
// Type definitions for @kiwicom/orbit-components
// Project: http://github.com/kiwicom/orbit-components

import * as React from "react";

import * as Common from "../common/common";

declare module "@kiwicom/orbit-components/lib/Text";

type Align = "left" | "center" | "right";
type Element = "p" | "span" | "div";
type Type =
  | "primary"
  | "secondary"
  | "attention"
  | "info"
  | "success"
  | "warning"
  | "critical"
  | "white";
type Weight = "normal" | "bold";

export interface Props extends Common.Global, Common.SpaceAfter {
  readonly type?: Type;
  readonly size?: Common.Size;
  readonly weight?: Weight;
  readonly align?: Align;
  readonly italic?: boolean;
  readonly uppercase?: boolean;
  readonly element?: Element;
  readonly children: React.ReactNode;
  readonly id?: string;
}

declare const Text: React.FunctionComponent<Props>;
export { Text, Text as default };
