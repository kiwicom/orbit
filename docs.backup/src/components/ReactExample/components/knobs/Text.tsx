import React from "react";
import { Text, InputField } from "@kiwicom/orbit-components";

import { Knob } from "../../Example";

interface Props extends Pick<Knob, "name"> {
  value: string;
  onChange: React.ChangeEventHandler<HTMLInputElement>;
}

const TextKnob = ({ value, name, onChange }: Props) => {
  return (
    <>
      <Text size="small">{name.split("-")[1]}</Text>
      <InputField name={name} value={value} onChange={onChange} />
    </>
  );
};

export default TextKnob;
