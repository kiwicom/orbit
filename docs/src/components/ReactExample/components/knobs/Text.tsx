import React from "react";
import { Text, InputField } from "@kiwicom/orbit-components";

import { Knob } from "../../Example";

interface Props extends Pick<Knob, "name"> {
  value: string;
  onChange: React.ChangeEventHandler<HTMLInputElement>;
}

const TextKnob = ({ value, name, onChange }: Props) => {
  return (
    <>
      <Text weight="bold" size="small" type="secondary">
        {name.split("-")[1]}
      </Text>
      {/* @ts-expect-error: todo */}
      <InputField name={name} value={value} onChange={onChange} />
    </>
  );
};

export default TextKnob;
