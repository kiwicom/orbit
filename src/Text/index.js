// @flow
import * as React from "react";
import * as tokens from "orbit-design-token";

const colorText = {
  primary: tokens.colorTextPrimary,
  secondary: tokens.colorTextSecondary,
  attention: tokens.colorTextAttention,
};

const weightText = {
  regular: tokens.fontWeightNormal,
  bold: tokens.fontWeightBold,
};

const sizeText = {
  large: tokens.fontSizeTextLarge,
  normal: tokens.fontSizeTextNormal,
  small: tokens.fontSizeTextSmall,
};

type Props = {
  type: $Keys<typeof colorText>,
  size: $Keys<typeof sizeText>,
  weight: $Keys<typeof weightText>,
  align: "left" | "center" | "right",
  italic?: boolean,
  uppercase?: boolean,
  element: "p" | "span" | "div",
  className?: string,
  children: React.Node,
};

function resolveScopedStyles(scope) {
  return {
    className: scope.props.className,
    styles: scope.props.children,
  };
}

const Text = (props: Props) => {
  const scoped = resolveScopedStyles(
    <scope>
      <style jsx>
        {`
          font-family: ${tokens.fontFamily};
          font-size: ${sizeText[props.size]};
          font-weight: ${weightText[props.weight]};
          color: ${colorText[props.type]};
          line-height: ${tokens.lineHeightText};
          text-align: ${props.align};
          ${props.uppercase ? `text-transform: uppercase` : ""};
          ${props.italic ? `font-style: italic` : ""};
        `}
      </style>
    </scope>,
  );

  return (
    <React.Fragment>
      {React.createElement(
        props.element,
        {
          className: [scoped.className, props.className].join(" "),
        },
        props.children,
      )}
      {scoped.styles}
    </React.Fragment>
  );
};

Text.defaultProps = {
  type: "primary",
  size: "normal",
  weight: "regular",
  align: "left",
  element: "p",
  uppercase: false,
  italic: false,
};

export default Text;
