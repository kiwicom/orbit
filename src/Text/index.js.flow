// @flow
/*
  DOCUMENTATION: https://orbit.kiwi/components/text/
*/
import type { ReactComponentStyled } from "styled-components";

import type { spaceAfter } from "../common/getSpacingToken/index";
import type { Globals } from "../common/common.js.flow";

type Align = "left" | "center" | "right";
type Element = "p" | "span" | "div";
type Type =
  | "primary"
  | "secondary"
  | "attention"
  | "info"
  | "success"
  | "warning"
  | "critical"
  | "white";
type Size = "large" | "normal" | "small";
type Weight = "normal" | "bold";

export type Props = {|
  +type?: Type,
  +size?: Size,
  +weight?: Weight,
  +align?: Align,
  +italic?: boolean,
  +uppercase?: boolean,
  +element?: Element,
  +children: React$Node,
  ...Globals,
  ...spaceAfter,
|};

declare export default React$ComponentType<Props>;

declare export var StyledText: ReactComponentStyled<any>;
