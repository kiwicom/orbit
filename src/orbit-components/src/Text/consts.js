// @flow
export const TYPE_OPTIONS = {
  PRIMARY: "primary",
  SECONDARY: "secondary",
  ATTENTION: "attention",
  INFO: "info",
  SUCCESS: "success",
  WARNING: "warning",
  CRITICAL: "critical",
  WHITE: "white",
};

export const SIZE_OPTIONS = {
  SMALL: "small",
  NORMAL: "normal",
  LARGE: "large",
};

export const WEIGHT_OPTIONS = {
  NORMAL: "normal",
  BOLD: "bold",
};

export const ALIGN_OPTIONS = {
  LEFT: "left",
  CENTER: "center",
  RIGHT: "right",
};

export const ELEMENT_OPTIONS = {
  P: "p",
  SPAN: "span",
  DIV: "div",
};
