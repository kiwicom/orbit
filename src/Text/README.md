# Text
To implement Text component into your project you'll need to add the import:
```jsx
import Text from "@kiwicom/orbit-components/lib/Text";
```
After adding import into your project you can use it simply like:
```jsx
<Text>Hello World!</Text>
```
## Props
Table below contains all types of the props available in the Text component.

| Name          | Type            | Default     | Description                      |
| :------------ | :-------------- | :---------- | :------------------------------- |
| align         | [`enum`](#enum) | `"left"`    | The align of the Text.
| children      | `React.Node`    |             | The content of the Text.
| dataTest      | `string`        |             | Optional prop for testing purposes.
| element       | [`enum`](#enum) | `"p"`       | The element used for the root node.
| italic        | `boolean`       | `false`     | If `true`, the Text will be in italic style.
| **size**      | [`enum`](#enum) | `"normal"`  | The size of the Text.
| spaceAfter    | `enum`          |             | Additional `margin-bottom` after component. [See this docs](https://github.com/kiwicom/orbit-components/tree/master/src/common/getSpacingToken)
| **type**      | [`enum`](#enum) | `"primary"` | The color type of the Text.
| uppercase     | `boolean`       | `false`     | If `true`, the Text will be in uppercase style.
| **weight**    | [`enum`](#enum) | `"regular"` | The weight of the Text.

### enum

| type            | align      | element  | size       | weight      |
| :-------------- | :--------- | :------- | :--------- | :---------- |
| `"attention"`   | `"left"`   | `"p"`    | `"small"`  | `"regular"` |
| `"primary"`     | `"center"` | `"span"` | `"normal"` | `"bold"`    |
| `"secondary"`   | `"right"`  | `"div"`  | `"large"`  |             |
| `"info"`        |            |          |            |             |
| `"success"`     |            |          |            |             |
| `"warning"`     |            |          |            |             |
| `"critical"`    |            |          |            |             |
| `"white"`       |            |          |            |             |
