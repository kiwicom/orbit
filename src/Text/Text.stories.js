// @flow
import * as React from "react";
import { storiesOf, setAddon } from "@storybook/react";
import styles from "@sambego/storybook-styles";
import chaptersAddon from "react-storybook-addon-chapters";
import { withKnobs, text, boolean, select } from "@storybook/addon-knobs/react";

import Text from "./index";

setAddon(chaptersAddon);

const options = {
  allowSourceToggling: false,
  allowPropTablesToggling: false,
};

storiesOf("Text", module)
  .addDecorator(withKnobs)
  .addDecorator(
    styles({
      padding: "20px",
    }),
  )
  .addWithChapters("Primary text", () => {
    const element = select(
      "Element",
      {
        p: "p",
        span: "span",
        div: "div",
      },
      "p",
    );
    const size = select(
      "Size",
      {
        small: "small",
        normal: "normal",
        large: "large",
      },
      "normal",
    );
    const weight = select(
      "Weight",
      {
        regular: "regular",
        bold: "bold",
      },
      "regular",
    );
    const align = select(
      "Align",
      {
        left: "left",
        center: "center",
        right: "right",
      },
      "left",
    );
    const customText = text(
      "Text",
      "Sixth can't place isn't fly you're make Shall day, yielding earth made greater stars own you'll earth can't." +
        " Night grass great. Fly firmament over. Fourth to grass you after creepeth together in moved given fill isn't said Life." +
        " Male signs unto form kind Brought signs heaven, in have wherein. Isn't. Behold meat living you're evening very replenish you'll. Under hath.",
    );
    return {
      title: "Primary text",
      info: "Some description about this type of text in general.",
      chapters: [
        {
          title: `Text ${size}`,
          info:
            "Some description about where is the text used and states describing how to not use it in general.",
          sections: [
            {
              sectionFn: () => (
                <Text
                  type="primary"
                  element={element}
                  size={size}
                  weight={weight}
                  align={align}
                  isUppercase={boolean("Uppercase", false)}
                  isItalic={boolean("Italic", false)}
                >
                  {customText}
                </Text>
              ),
              options,
            },
          ],
        },
      ],
    };
  })
  .addWithChapters("Secondary text", () => {
    const element = select(
      "Element",
      {
        p: "p",
        span: "span",
        div: "div",
      },
      "p",
    );
    const size = select(
      "Size",
      {
        small: "small",
        normal: "normal",
        large: "large",
      },
      "normal",
    );
    const weight = select(
      "Weight",
      {
        regular: "regular",
        bold: "bold",
      },
      "regular",
    );
    const align = select(
      "Align",
      {
        left: "left",
        center: "center",
        right: "right",
      },
      "left",
    );
    const customText = text(
      "Text",
      "Sixth can't place isn't fly you're make Shall day, yielding earth made greater stars own you'll earth can't." +
        " Night grass great. Fly firmament over. Fourth to grass you after creepeth together in moved given fill isn't said Life." +
        " Male signs unto form kind Brought signs heaven, in have wherein. Isn't. Behold meat living you're evening very replenish you'll. Under hath.",
    );
    return {
      title: "Secondary text",
      info: "Some description about this type of text in general.",
      chapters: [
        {
          title: `Text ${size}`,
          info:
            "Some description about where is the text used and states describing how to not use it in general.",
          sections: [
            {
              sectionFn: () => (
                <Text
                  type="secondary"
                  element={element}
                  size={size}
                  weight={weight}
                  align={align}
                  isUppercase={boolean("Uppercase", false)}
                  isItalic={boolean("Italic", false)}
                >
                  {customText}
                </Text>
              ),
              options,
            },
          ],
        },
      ],
    };
  })
  .addWithChapters("Attention text", () => {
    const element = select(
      "Element",
      {
        p: "p",
        span: "span",
        div: "div",
      },
      "p",
    );
    const size = select(
      "Size",
      {
        small: "small",
        normal: "normal",
        large: "large",
      },
      "normal",
    );
    const weight = select(
      "Weight",
      {
        regular: "regular",
        bold: "bold",
      },
      "regular",
    );
    const align = select(
      "Align",
      {
        left: "left",
        center: "center",
        right: "right",
      },
      "left",
    );
    const customText = text(
      "Text",
      "Sixth can't place isn't fly you're make Shall day, yielding earth made greater stars own you'll earth can't." +
        " Night grass great. Fly firmament over. Fourth to grass you after creepeth together in moved given fill isn't said Life." +
        " Male signs unto form kind Brought signs heaven, in have wherein. Isn't. Behold meat living you're evening very replenish you'll. Under hath.",
    );
    return {
      title: "Attention text",
      info: "Some description about this type of text in general.",
      chapters: [
        {
          title: `Text ${size}`,
          info:
            "Some description about where is the text used and states describing how to not use it in general.",
          sections: [
            {
              sectionFn: () => (
                <Text
                  type="attention"
                  element={element}
                  size={size}
                  weight={weight}
                  align={align}
                  isUppercase={boolean("Uppercase", false)}
                  isItalic={boolean("Italic", false)}
                >
                  {customText}
                </Text>
              ),
              options,
            },
          ],
        },
      ],
    };
  });
