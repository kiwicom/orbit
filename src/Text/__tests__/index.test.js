// @flow
import * as React from "react";
import { shallow } from "enzyme";

import Text from "../";

describe("Rendered with children", () => {
  it("Should contain text", () => {
    const text = "Children text";
    const component = shallow(
      <Text type="primary" size="small">
        {text}
      </Text>,
    );
    expect(component).toMatchSnapshot();
  });
});
