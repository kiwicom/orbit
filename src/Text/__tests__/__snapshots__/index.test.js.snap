// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendered with children Should contain text 1`] = `
<React.Fragment>
  <p
    className="jsx-487647055 "
  >
    Children text
  </p>
  <JSXStyle
    css=".__jsx-style-dynamic-selector{font-family:\\"Roboto\\", -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif;font-size:12px;font-weight:400;color:#46515e;line-height:1.4;text-align:left;;;}"
    dynamic={
      Array [
        "\\"Roboto\\", -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif",
        "12px",
        "400",
        "#46515e",
        "1.4",
        "left",
        "",
        "",
      ]
    }
    styleId="143391915"
  />
</React.Fragment>
`;
